### PR TITLE
remove _toString, always use NAME_KEY

### PR DESCRIPTION
**Pull request by stefanpenner**
no longer:
- have two concepts
- processAllNamespaces + find NAME_KEY if the name was pre-determined by the container (previously _toString was set, but not NAME_KEY, which made this code assume it must calculate the name again)
- change shape of `this` with a constant descriptor of toString
- have _toString (NAME_KEY is now used)

---

I noticed ember-data's test suite was spending an unexpected amount of time in processAllNamespaces, which lead me to realize `_toString` is not consider  in classToString, but it really should have been.

### CODE REVIEW

**Comment by stefanpenner** [comment]
@krisselden r?

File: packages/container/lib/container.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { dictionary, symbol, setOwner, OWNER } from 'ember-utils';
 import { ENV } from 'ember-environment';
-import { assert, deprecate, runInDebug } from 'ember-metal';
+import { assert, deprecate, runInDebug, NAME_KEY } from 'ember-metal';
```
Comment (rwjblue) [inline packages/container/lib/container.js]: Can you move `NAME_KEY` (which I think is just a symbol right?) into the `ember-utils` package instead?  We are ultimately trying to make `container` package not rely on ember-metal (the `assert`, `deprecate`, `runInDebug` functionality was the only remaining bit).

Comment (stefanpenner) [inline packages/container/lib/container.js]: sure

Comment (stefanpenner) [inline packages/container/lib/container.js]: i made the change, but something after moving that stuff around has broken things ... need to figure out what

**Comment by rwjblue** [comment]
This looks good to me...

**Pull request merged**